packages/syft/src/syft/types/syft_metaclass.py

```diff
@@ -29,11 +30,28 @@ class Empty(metaclass=EmptyType):
 
 class PartialModelMetaclass(ModelMetaclass):
     def __call__(cls: type[_T], *args: Any, **kwargs: Any) -> _T:
-        for field_info in cls.model_fields.values():
-            if field_info.annotation is not None and field_info.is_required():
-                field_info.annotation = field_info.annotation | EmptyType
-                field_info.default = Empty
+        orig_field_map: dict = defaultdict(dict)
+
+        def optionalize(restore: bool = False) -> None:
+            if not restore:
+                for field_name, field_info in cls.model_fields.items():
+                    if field_info.annotation is not None and field_info.is_required():
+                        orig_field_map[field_name]["annotation"] = field_info.annotation
+                        orig_field_map[field_name]["default"] = field_info.default
+                        field_info.annotation = field_info.annotation | EmptyType
+                        field_info.default = Empty
+            else:
+                for field_name, field_info in cls.model_fields.items():
+                    if field_info.default == Empty:
+                        field_info.annotation = orig_field_map[field_name]["annotation"]
+                        field_info.default = orig_field_map[field_name]["default"]
+
+        optionalize()
+        cls.model_rebuild(force=True)
```

Comment:
No lock at this point causes issues with multithreading.

Reply:
Earlier we use to lock during init but now we don't have control over it.

Reply:
Resolved by: [f8b40af](https://github.com/OpenMined/PySyft/pull/8642/commits/f8b40af31672e56e834a56142956b906374f94b4)